test/test_euler.py

```diff
@@ -57,8 +57,9 @@
 logger = logging.getLogger(__name__)
 
 
+@pytest.mark.parametrize("nspecies", [1, 2, 3])
 @pytest.mark.parametrize("dim", [1, 2, 3])
```

Comment:
Do we really need to test all nine combinations?

Reply:
Perhaps not, but I perceive that I should test at least nspec x ndim = [1, 10] x [1, 2, 3].

Reply:
I cut down on the number of tests some, but I at least perceive that the remainder of them are needed.  I have the feeling this is not what you wanted.